phabfive/diffusion.py

```diff
@@ -37,13 +37,50 @@ def _validate_credential_type(self, credential):
                     "token",
                 ):
                     raise PhabfiveDataException(
-                        "{0} is not type of ssh-generated-key, ssh-key-text or token.".format(
-                            credential
+                        "{0} is not type of 'ssh-generated-key', 'ssh-key-text' or 'token' but type '{1}'".format(
+                            credential[credential_phid]["monogram"],
+                            credential[credential_phid]["type"],
                         )
                     )
 
         return credential_phid
 
+    def get_object_identifier(self, repo_name=None, uri_name=None):
+        """Phabfive wrapper that connects to Phabricator and identify repository or uri object_identifier.
+
+        :type repo_name: str
+        :type uri_name: str
+
+        :rtype object_identifier: str
+        """
+        object_identifier = ""
+        repos = self.get_repositories(attachments={"uris": True})
+
+        for repo in repos:
+            name = repo["fields"]["shortName"]
+            if repo_name == name:
```

Comment:
Having these extremly nested loops with validation checks is not pretty code. If possible in the conditions that is possible to check if you should continue or not, just do that check instead and use a `continue` block instead.

For example on this row, you can change it to 
```
if repo_name != name:
    continue
```
and reindent the rest of the code one step to the left. The same thing goes for all of the other if checks further into the loop, if possible just use continue.

Reply:
True true, on it